org.mixedrealitytoolkit.input/Tracking/HandPoseDriver.cs

```diff
@@ -0,0 +1,266 @@
+        /// Sets the transform that is being driven by the <see cref="TrackedPoseDriver"/>. This will only set requested values, but regardless of tracking state.
+        /// </summary>
+        /// <param name="newPosition">The new local position to possibly set.</param>
+        /// <param name="newRotation">The new local rotation to possibly set.</param>
+        protected virtual void ForceSetLocalTransform(Vector3 newPosition, Quaternion newRotation)
+        {
+#if HAS_SET_LOCAL_POSITION_AND_ROTATION
+            if (trackingType == TrackingType.RotationAndPosition)
+            {
+                transform.SetLocalPositionAndRotation(newPosition, newRotation);
+                return;
+            }
+#endif
+
+            if (trackingType == TrackingType.RotationAndPosition ||
+                trackingType == TrackingType.RotationOnly)
+            {
+                transform.localRotation = newRotation;
+            }
+
+            if (trackingType == TrackingType.RotationAndPosition ||
+                trackingType == TrackingType.PositionOnly)
+            {
+                transform.localPosition = newPosition;
+            }
+        }
+
+        /// <summary>
+        /// The base class hasn't made OnEnable virtual, so we need to bind to tracking state updates
+        /// in the first update. If base ever makes OnEnable virtual, we can move binding to OnEnabled.
+        ///  </summary>
+        private void OnFirstUpdate()
+        {
+            UnbindTrackingState();
+            BindTrackingState();
+            ForceTrackingStateUpdate();
+        }
+
+        /// <summary>
+        /// The base class has not made OnDisable virtual, so we need to check for disablement in
+        /// tracking state callbacks. If base ever make OnDisable virtual, we can unbind in OnDisable instead.
+        /// </summary>
+        private void HandleDisablement()
+        {
+            if (!isActiveAndEnabled || !Application.isPlaying)
+            {
+                UnbindTrackingState();
+            }
+        }
+
+        /// <summary>
+        /// Force an update of the tracking state from the Input Action Reference.
+        /// </summary>
+        private void ForceTrackingStateUpdate()
+        {
+            var trackingStateAction = trackingStateInput.action;
+            if (trackingStateAction != null && !trackingStateAction.enabled)
+            {
+                // Treat a disabled action as the default None value for the ReadValue call
+                m_trackingState = InputTrackingState.None;
+                return;
+            }
+
+            if (trackingStateAction == null || trackingStateAction.bindings.Count == 0)
+            {
+                // Treat an Input Action Reference with no reference the same as
+                // an enabled Input Action with no authored bindings, and allow driving the Transform pose.
+                m_trackingState = InputTrackingState.Position | InputTrackingState.Rotation;
```

Comment:
I'm not sure I follow this logic exactly. In this case, should we drive `m_trackingState` based on whether we're able to get a value from the polyfill?

Reply:
I was trying to reproduce logic used in `UnityEngine.InputSystem.XR.TrackedPoseDriver`... see this in `com.unity.inputsystem@1.8.1\InputSystem\Plugins\XR\TrackedPoseDriver.cs`

```cs
        void ReadTrackingState()
        {
            var trackingStateAction = m_TrackingStateInput.action;
            if (trackingStateAction != null && !trackingStateAction.enabled)
            {
                // Treat a disabled action as the default None value for the ReadValue call
                m_CurrentTrackingState = TrackingStates.None;
                return;
            }

            if (trackingStateAction == null || trackingStateAction.m_BindingsCount == 0)
            {
                // Treat an Input Action Reference with no reference the same as
                // an enabled Input Action with no authored bindings, and allow driving the Transform pose.
                m_CurrentTrackingState = TrackingStates.Position | TrackingStates.Rotation;
                return;
            }

           ....
```

My thought is that it would be odd if `HandPoseDriver` determined tracking state differently than the base component.  Also, it's super unfortunate that the base does not expose `m_CurrentTrackingState` to subclasses.

Reply:
Sounds good to me! Thanks for the explanation!